lib/thumbs/pull_request_worker.rb

```diff
@@ -1,47 +1,66 @@
 module Thumbs
   class PullRequestWorker
     attr_reader :build_dir
-    attr_reader :build_status
-    attr_accessor :build_steps
-    attr_reader :minimum_reviewers
     attr_reader :repo
     attr_reader :pr
-    attr_accessor :thumb_config
     attr_reader :log
     attr_reader :org
-    attr_accessor :client
-    attr_reader :pull_request_id
+    attr_reader :thumbs_config_default
 
     def initialize(options)
-      @repo = options[:repo]
-      @org = repo.split('/').first
-      @client = Octokit::Client.new(:netrc => true)
-      @pr = @client.pull_request(options[:repo], options[:pr])
-      @pull_request_id=get_pull_request_id
-      @build_dir=options[:build_dir] || "/tmp/thumbs/#{build_guid}"
-      persisted_build_status = read_build_status
-      @build_status = persisted_build_status || {:steps => {}}
-      @build_steps = []
+      if options[:url]
+        @repo, @pr = parse_github_pr_url(options[:url])
+      end
+      @repo ||= options[:repo]
+      @org = @repo.split('/').first
+      @pr ||= options[:pr]
+      @build_dir = options[:build_dir] || "/tmp/thumbs/#{build_guid}"
+      @thumbs_config_default = options[:thumbs_config_default]
+
       prepare_build_dir
-      load_thumbs_config
-      @minimum_reviewers = thumb_config && thumb_config.key?('minimum_reviewers') ? thumb_config['minimum_reviewers'] : 2
-      @timeout=thumb_config && thumb_config.key?('timeout') ? thumb_config['timeout'] : 1800
+    end
+
+    def reset_state
+      @pull_request =
+      @event =
+      @open_pull_requests =
+      @commits =
+      @base_commits =
+      @head_commits =
+      @comments_since_most_recent_commit =
+      nil
+    end
+
+    def octokit_client
+      @octokit_client ||= Octokit::Client.new(:netrc => true)
```

Comment:
This lazy initialization may be unnecessary, since caching the result is to reduce the cost of multiple costs, but I assume creating the client is beyond non-trivial, minimally reading the netrc file, so I figured it is helpful to expedite execution, testing, and irb usage.